codegen/src/utils/fetch_metadata.rs

```diff
@@ -12,14 +13,30 @@ use jsonrpsee::{
 };
 use std::time::Duration;
 
+/// The metadata version that is fetched from the node.
+#[derive(Default)]
+pub enum MetadataVersion {
+    /// Version 14.
+    #[default]
+    V14,
+    /// Latest unstable version of the metadata.
+    Unstable,
+}
```

Comment:
As per my earlier comment, I guess I'd also "future proof" this and just allow it to represent any possible version that may exist going forwards eg:

```rust
pub enum MetadataVersion {
    Stable(u32),
    Unstable
}
```

And then error only at the point at which the metadata obtained is seen to not be suitable (ie in the codegen when trying to decode and use it)